src/types/package.rs

```diff
@@ -450,6 +482,107 @@ pub struct PackageDescriptor {
     pub package_type: PackageType,
 }
 
+/// Describes a package descriptor in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct LockFilePackageDescriptor {
```

Comment:
We generally write "lockfile" as one word. This struct even writes it as one word within its own fields.